src/test/java/rx/internal/operators/OperatorSortedTest.java

```diff
@@ -0,0 +1,103 @@
+package rx.internal.operators;
+
+import org.junit.Test;
+import rx.Observable;
+import rx.Observer;
+import rx.functions.Func2;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+public class OperatorSortedTest {
+
+    @Test
+    public void testSortedList() {
+        Observable<Integer> w = Observable.just(1, 3, 2, 5, 4);
+        Observable<Integer> observable = w.sorted();
+
+        @SuppressWarnings("unchecked")
+        Observer<Integer> observer = mock(Observer.class);
+        observable.subscribe(observer);
+        verify(observer, times(1)).onNext(1);
+        verify(observer, times(1)).onNext(2);
+        verify(observer, times(1)).onNext(3);
+        verify(observer, times(1)).onNext(4);
+        verify(observer, times(1)).onNext(5);
+        verify(observer, never()).onError(any(Throwable.class));
+        verify(observer, times(1)).onCompleted();
+    }
+
+    @Test
+    public void testSortedListWithCustomFunction() {
+        Observable<Integer> w = Observable.just(1, 3, 2, 5, 4);
+        Observable<Integer> observable = w.sorted(new Func2<Integer, Integer, Integer>() {
+
+            @Override
+            public Integer call(Integer t1, Integer t2) {
+                return t2 - t1;
+            }
+
+        });
+
+        @SuppressWarnings("unchecked")
+        Observer<Integer> observer = mock(Observer.class);
+        observable.subscribe(observer);
+        verify(observer, times(1)).onNext(5);
+        verify(observer, times(1)).onNext(4);
+        verify(observer, times(1)).onNext(3);
+        verify(observer, times(1)).onNext(2);
+        verify(observer, times(1)).onNext(1);
+        verify(observer, never()).onError(any(Throwable.class));
+        verify(observer, times(1)).onCompleted();
+    }
+
+    @Test
+    public void testSortedCustomComparator() {
+        Observable<Integer> w = Observable.just(1, 3, 2, 5, 4);
+        Observable<Integer> observable = w.sorted(new Func2<Integer, Integer, Integer>() {
+            @Override
+            public Integer call(Integer t1, Integer t2) {
+                return t1.compareTo(t2);
+            }
+
+        });
+
+        @SuppressWarnings("unchecked")
+        Observer<Integer> observer = mock(Observer.class);
+        observable.subscribe(observer);
+        verify(observer, times(1)).onNext(1);
+        verify(observer, times(1)).onNext(2);
+        verify(observer, times(1)).onNext(3);
+        verify(observer, times(1)).onNext(4);
+        verify(observer, times(1)).onNext(5);
+        verify(observer, never()).onError(any(Throwable.class));
+        verify(observer, times(1)).onCompleted();
+    }
+
+    @Test
+    public void testSortedWithNonComparable() {
```

Comment:
Not sure about this test. if you want to remove it or do it differently please let me know